doc/12633_SymmetricAssociation.adoc

```diff
@@ -0,0 +1,106 @@
+
+=== 5 Design
+
+For a simple binary association, the Ciera architecture renders a 'selector' - a pointer or 
+set of pointers - in each of the participant classes which serve to relate instances. Each 
+relationship also manages, in an relationship-specific extent, a collection of Relationship 
+instances, each of which maintains a pair of instance identifiers for two participant instances.
+
+For reflexive associations there are two distinct cases to be addressed:
+
+==== 5.1 Simple Binary Reflexive
+
+Binary reflexive associations are supported using two selectors in the class, named with the differing role phrases.
+This supports the concept of 'direction' in the establishment and navigation of the relationship. In the 
+symmetric case it is a minimal change to suppress the rendering of the second selector, allowing the 
+linkage between instances to become symmetric. The removal of Relationship instances from the 
+relationship extent is slightly problematic: the ordering of the two participant IDs is significant. 
+Hence, the unrelate logic is modified to try both possible orderings of the ID pair.
+
+==== 5.1 Associative Reflexive
+
+In the asymmetric associative reflexive case Ciera breaks the linked association into a pair of simple binary 
+relationships between the participant 'reflexed' class and the associative link class,  
+sharing the same relationship number but differing by role phrase, which, again, support the concept 
+of direction. Note that the associative class maintains a selector for each of the contributing 
+binary relationships. For the symmetric case only one relationship is maintained, however the selector 
+in the associative class is made multi-valued regardless of original association multiplicity. An additional 
+peer-to-peer selector is generated for the 'reflexed' class. The implementation of this peer-to-peer selector 
+navigates via the associative class, returning all values from the associative class selector, but with the 
+originating instance removed. The relationship extent management is not an issue: the ID ordering is always 
+from the associative class to the 'reflexed' class and, hence, the removal of Relationship instances is not 
+affected.
+
+=== 6 Work required
+
+==== 6.1 Simple binary reflexive
+
+* Transform of selectors is modified to avoid generating two selectors for the participant class.
+* Transformation of Relate and Unrelate statements are modified to use the single role phrase.
+* Modification of the unrelate logic to try both attempts at removal of Relationship instances.
+
+==== 6.2 Associative reflexive
+
+* At transform, the association is not split into two relationships.
+* The association chain link is not resolved into a pair of links to and from the associative class.
+* Transform of selectors is modified to generate a peer-to-peer selector for the 'reflexed' class.
+* No storage is allocated for the peer-to-peer selector; it is used only for navigation.
```

Comment:
Wording seems odd here.  Selectors are always used only for navigation, right?  There is storage (a pointer) in each associated instance referring to the linked associative instance.  But, the peer-to-peer selector requires no additional storage in the associated instance because the associative instance contains a set pointing to both associated instances. The peer-to-peer selector is just a method containing the intelligence to navigate from one associated instance to the other, using the references described above.  Is this true?